editor/components/inner-blocks/index.js

```diff
@@ -43,15 +43,19 @@ class InnerBlocks extends Component {
 	updateNestedSettings() {
 		const {
 			blockListSettings,
-			allowedBlocks: nextAllowedBlocks,
+			allowedBlocks,
+			templateLock,
+			parentLock,
 			updateNestedSettings,
 		} = this.props;
 
-		const allowedBlocks = get( blockListSettings, [ 'allowedBlocks' ] );
-		if ( ! isShallowEqual( allowedBlocks, nextAllowedBlocks ) ) {
-			updateNestedSettings( {
-				allowedBlocks: nextAllowedBlocks,
-			} );
+		const newSettings = {
+			allowedBlocks,
+			templateLock: templateLock === undefined ? parentLock : templateLock,
```

Comment:
Should it be the responsibility of someone rendering a `BlockList` to pass the `parentLock`, or should that component be smart enough to inherit?

If a parent block has locking but the child does not, which one wins?

Reply:
When rendering a BlockList we don't pass the parentLock. We pass the templateLock. parentLock is just used for the case where templateLock is undefined and in that case we store parentLock in templateLock. Besides that case we never use parentLock again. 

> If a parent block has locking but the child does not, which one wins?

If the locking of an InnerBlocks area is undefined we use the locking of the parent. In all other cases, we use the templateLock passed even if the locking of the child is more permissive than the locking of the father.
Before we used always the less permissive locking of the tree https://github.com/WordPress/gutenberg/pull/6991#discussion_r191383168 but it was changed.